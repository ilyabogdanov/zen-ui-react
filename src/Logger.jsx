
const LEVEL = {
    OFF     : 0,
    WARN    : 1,
    INFO    : 2,
    DEBUG   : 3,
    TRACE   : 4
};
const Logger = {
    level: LEVEL.TRACE,
    error (e) {
        throw new Error(e);
    },
    warn () {
        if (this.level >= LEVEL.WARN) {
            try {
                console.warn.apply(this, ...arguments)
            } catch (e) {
                console.warn.apply(this, arguments)
            }
        }
    },
    info () {
        if (this.level >= LEVEL.INFO) {
            try {
                console.warn.apply(this, ...arguments)
            } catch (e) {
                console.warn.apply(this, arguments)
            }
        }
    },
    debug () {
        if (this.level >= LEVEL.DEBUG) {
            try {
                console.log.apply(this, ...arguments)
            } catch (e) {
                console.log.apply(this, arguments)
            }
        }
    },
    trace () {
        if (this.level >= LEVEL.TRACE) {
            try {
                console.log.apply(this, ...arguments);
            } catch (e) {
                console.log.apply(this, arguments)
            }
        }
    }
};
export default Logger;
export { LEVEL };
