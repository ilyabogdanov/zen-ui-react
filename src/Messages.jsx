
const _VALID_BOOLEAN_VALUE_ = "either true, false, or empty attribute which equals true";
const _VALID_PIXEL_VALUE_ = "either (a) Natural number, starting with anything but zero, ending with \"px\", or (b) Zero, without \"px\"";
const _BAD_HEIGHT_ = "\"height\" must be either (1) undefined, (2) 100%, or (3) valid pixel value, which means "+_VALID_PIXEL_VALUE_+".";
const _READONLY_IS_DISABLED_ = "will be forced to disabled because it is readonly.";

const COMMON_MESSAGES = {
    BAD_PADDING: "Property \"padding\" must be either 1,2 or 4 values, each of which must be valid pixel value, which means "+_VALID_PIXEL_VALUE_+"."
};
const TEXT_MESSAGES = {
    BAD_CLIPPED: "Property \"clipped\" must be but is not valid boolean value, which means "+_VALID_BOOLEAN_VALUE_+"."
};
const CONJOINED_BUTTONS_MESSAGES = {
    INVALID_CHILD: "Conjoined buttons container may contain only these components: push button, link button, toggle button, toggle radio button.",
    NO_CHILDREN: "Conjoined buttons container must have children"
};
const CHECKBOX_MESSAGES = {
    READONLY_IS_DISABLED: "Checkbox "+_READONLY_IS_DISABLED_
};
const RADIO_MESSAGES = {
    READONLY_IS_DISABLED: "Radio "+_READONLY_IS_DISABLED_,
    INVALID_CHILD: "Radio group may contain only Radio component or arrays of them"
};
const HORIZONTAL_LAYOUT_MESSAGES = {
    BAD_HEIGHT: "Horizontal layout's "+_BAD_HEIGHT_,
    ILLEGAL_CHILD: "Horizontal layout may have only columns or arrays of them.",
    NO_CHILDREN: "Horizontal layout must have at least one column.",
    STRETCH_WITH_HEIGHT: "Horizontal layout must not have both \"stretch\" and \"height\" properties.",
    EXCESSIVE_WIDTH: "Total width of all columns must not be more than 100%;",
    INSUFFICIENT_WIDTH: "Either (a) total width of all columns must be 100%, or (b) if less, then ratio may be used to divide remaining space.",
    NO_SPACE_FOR_RATIO: "No free space to accommodate columns with ratio width.",
    INVALID_WIDTH: "Horizontal layout column \"width\" must be either (a) pixel value, (b) percent value, or (c) ratio, which means Natural number.",
    NO_WIDTH: "Horizontal layout column must have \"width\".",
    COLUMN_WIDTH_REQUIRED: "Column with must be set.",
    TOTAL_WIDTH_LIMIT_EXCEEDED: "Total width of all columns, percent and ration, does not equal precisely 100%. It should not cause trouble but be careful anyway.",
    NO_SPACE_FOR_NO_WIDTH: "No free space to accommodate columns with undefined width."
};
const VERTICAL_LAYOUT_MESSAGES = {
    BAD_HEIGHT: "Vertical layout's "+_BAD_HEIGHT_,
    ILLEGAL_CHILD: "Vertical layout may have only rows or arrays of them.",
    NO_CHILDREN: "Vertical layout must have at least one row.",
    MULTIPLE_STRETCH_ROWS: "Vertical layout may have only one stretch row."
};
const WINDOW_MESSAGES = {
    ILLEGAL_CHILD: "Window can't have illegal children",
    NO_CHILDREN: "Window must have head and body"
};
const WINDOW_HEAD_MESSAGES = {
    ILLEGAL_CHILD: "Window head can have only close button"
};
const UTILS_MESSAGES = {
    NO_STYLE_DEST: "style destination must be provided"
};
const TREE_MESSAGES = {
    ILLEGAL_CHILD: "Illegal child, see docs for more info."
};
export {
    COMMON_MESSAGES,
    CONJOINED_BUTTONS_MESSAGES,
    CHECKBOX_MESSAGES,
    RADIO_MESSAGES,
    TEXT_MESSAGES,
    HORIZONTAL_LAYOUT_MESSAGES,
    VERTICAL_LAYOUT_MESSAGES,
    WINDOW_MESSAGES,
    WINDOW_HEAD_MESSAGES,
    TREE_MESSAGES,
    UTILS_MESSAGES
};
