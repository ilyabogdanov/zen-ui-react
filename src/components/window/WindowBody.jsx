import React from "react";

/**
 * @class
 * @classdesc Window body
 * @author Ilya Bogdanov <public@ilyabogdanov.ru>
 * @license Apache-2.0
 *
 * @prop {string} props.padding
 * @prop {boolean} props.stretch
 *
 * @prop {*} props.children
 */
export default class WindowBody extends React.Component {
    render() {
        const { stretch, padding, children } = this.props;
        return (
            <div className="zen_ui__window_body" style={stretch!==true ? { padding: padding } : {}}>
                <div style={stretch===true ? { margin: padding } : {}}>
                    {children}
                </div>
            </div>
        );
    }
};