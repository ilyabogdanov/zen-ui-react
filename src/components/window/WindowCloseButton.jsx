import React from "react";

/**
 * @class
 * @classdesc Window close button
 * @author Ilya Bogdanov <public@ilyabogdanov.ru>
 * @license Apache-2.0
 *
 * @prop {object} props
 * @prop {string} props.palette
 * @prop {function} props.onClick
 *
 * @extends React.Component
 */
export default class WindowCloseButton extends React.Component {
    constructor() {
        super();
        this._onClick = this._onClick.bind(this);
    }
    _onClick() {
        this.props.onClick();
    }
    render() {
        const { _onClick } = this;
        const { palette } = this.props;
        return (
            <div className="zen_ui__horizontal_layout_column">
                <div style={{width: "2.3rem", padding: "0.08rem 0.08rem 0.16rem 0.08rem"}}>
                    <div>
                        <div className="zen_ui__button_container" data-stretch="true">
                            <button className="zen_ui__button" data-palette={palette} onClick={_onClick}>
                                <div className="zen_ui__vector_icon__close"></div>
                            </button>
                            <div></div>
                        </div>
                    </div>
                </div>
            </div>
        );
    }
};
