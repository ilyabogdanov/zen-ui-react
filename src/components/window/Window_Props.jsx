
const Window_Props = {
    palette: {
        BRANDED: "branded",
        ORDINARY: "ordinary"
    }
};
export default Window_Props;