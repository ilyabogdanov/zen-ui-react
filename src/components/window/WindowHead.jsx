import React from "react";
import WindowCloseButton from "./WindowCloseButton";
import Logger from "../../Logger";
import { WINDOW_HEAD_MESSAGES } from "../../Messages";

/**
 * @class
 * @classdesc Window head
 * @author Ilya Bogdanov <public@ilyabogdanov.ru>
 * @license Apache-2.0
 *
 * @prop {object} props
 * @prop {string} props.title
 * @prop {string} props.palette
 *
 * @prop {*} props.children
 */
export default class WindowHead extends React.Component {
    render() {
        const { title, palette, children } = this.props;

        const __children__ = React.Children.map(children, (e) => {
            if (e.type===WindowCloseButton) {
                return React.cloneElement(e, {palette: palette});
            } else {
                Logger.error(WINDOW_HEAD_MESSAGES.ILLEGAL_CHILD);
            }
        });
        return (
            <div className="zen_ui__window_head">
                <div className="zen_ui__stretch_margin">
                    <div className="zen_ui__horizontal_layout" style={{height:"100%"}}>
                        <div className="zen_ui__horizontal_layout_column" style={{width:"100%"}}>
                            <div>
                                <div>
                                    <span className="zen_ui__clipped_text">
                                        <span style={{ padding: "0 0.7rem" }}>{title}</span>
                                    </span>
                                </div>
                            </div>
                        </div>
                        {__children__}
                    </div>
                </div>
            </div>
        );
    }
};