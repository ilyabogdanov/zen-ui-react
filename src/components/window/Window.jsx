import React from "react";
import WindowHead from "./WindowHead";
import WindowBody from "./WindowBody";
import Window_Props from "./Window_Props";
import Logger from "../../Logger";
import { WINDOW_MESSAGES } from "../../Messages";

/**
 * @class
 * @classdesc Window
 * @author Ilya Bogdanov <public@ilyabogdanov.ru>
 * @license Apache-2.0
 *
 * @prop {object} props
 * @prop {boolean} props.stretch
 * @prop {string} props.palette
 * @prop {string} props.width
 * @prop {string} props.minWidth
 * @prop {string} props.maxWidth
 *
 * @prop {*} props.children
 */
export default class Window extends React.Component {
    render() {
        const { stretch, palette, width, maxWidth, minWidth, children } = this.props;

        let hasHead = false;
        let hasBody = false;
        const __children__ = React.Children.map(children, (e) => {
            if (e.type===WindowHead) {
                hasHead=true;
                return React.cloneElement(e, {palette: palette ? palette : Window_Props.palette.ORDINARY});
            } else if (e.type===WindowBody) {
                hasBody=true;
                return React.cloneElement(e, {stretch: stretch});
            } else {
                Logger.error(WINDOW_MESSAGES.ILLEGAL_CHILD);
            }
        });
        (!hasHead || !hasBody) && Logger.error(WINDOW_MESSAGES.NO_CHILDREN);

        return (
            <div className="zen_ui__window"
                 style={{
                     width: width ? width : null,
                     minWidth: minWidth ? minWidth : null,
                     maxWidth: maxWidth ? maxWidth : null
                 }}
                 data-stretch={stretch===true}
                 data-palette={palette ? palette : Window_Props.palette.ORDINARY}>
                {__children__}
            </div>
        );
    }
};