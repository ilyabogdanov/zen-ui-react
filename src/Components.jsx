import React from "react";

/**
 * button
 */
import Button_Props           from "./components/form/Button_Props";
import Combobox               from "./components/form/Combobox";
import ConjoinedButtons       from "./components/form/ConjoinedButtons";
import LinkButton             from "./components/form/LinkButton";
import PushButton             from "./components/form/PushButton";
import ToggleButton           from "./components/form/ToggleButton";
import ToggleRadioButton            from "./components/form/ToggleRadioButton";

/**
 * content
 */
import Text                   from "./components/content/Text";
import Text_Props             from "./components/content/Text_Props";

/**
 * input
 */
import Checkbox               from "./components/form/Checkbox";
import Checkbox_Props         from "./components/form/Checkbox_Props";
import NumberField            from "./components/form/NumberField";
import PasswordField          from "./components/form/PasswordField";
import Radio                  from "./components/form/Radio";
import TextField              from "./components/form/TextField";
import TimeField              from "./components/form/TimeField";

/**
 * layout
 */
import HorizontalLayout       from "./components/layout/HorizontalLayout";
import HorizontalLayoutColumn from "./components/layout/HorizontalLayoutColumn";
import VerticalLayout         from "./components/layout/VerticalLayout";
import VerticalLayoutRow      from "./components/layout/VerticalLayoutRow";

/**
 * navigation
 */
import HorizontalMenu         from "./components/navigation/HorizontalMenu";
import HorizontalMenuImage    from "./components/navigation/HorizontalMenuImage";
import HorizontalMenuLink     from "./components/navigation/HorizontalMenuLink";
import HorizontalMenuColumn   from "./components/navigation/HorizontalMenuColumn";
import Path                   from "./components/navigation/Path";
import PathItem               from "./components/navigation/PathItem";

/**
 * panel
 */
import LoadingMask            from "./components/panel/LoadingMask";
import Tree                   from "./components/panel/Tree";
import TreeBranch             from "./components/panel/TreeBranch";
import TreeBranchlet          from "./components/panel/TreeBranchlet";
import TreeLeaf               from "./components/panel/TreeLeaf";
import TreeLeafColumn         from "./components/panel/TreeLeafColumn";
import TreeLinkButton         from "./components/panel/TreeLinkButton";
import TreePushButton         from "./components/panel/TreePushButton";

/**
 * window
 */
import Modal                  from "./components/window/Modal";
import Window                 from "./components/window/Window";
import Window_Props           from "./components/window/Window_Props";
import WindowBody             from "./components/window/WindowBody";
import WindowCloseButton      from "./components/window/WindowCloseButton";
import WindowHead             from "./components/window/WindowHead";

/**
 * Container
 */
import ScrollArea                  from "./components/container/ScrollArea";

const Zen = {
    Button_Props: Button_Props,
    Combobox: Combobox,
    ConjoinedButtons: ConjoinedButtons,
    LinkButton: LinkButton,
    PushButton: PushButton,
    ToggleButton: ToggleButton,
    ToggleRadioButton: ToggleRadioButton,
    Text: Text,
    Text_Props: Text_Props,
    Checkbox: Checkbox,
    Checkbox_Props: Checkbox_Props,
    NumberField: NumberField,
    PasswordField: PasswordField,
    Radio: Radio,
    TextField: TextField,
    TimeField: TimeField,
    HorizontalLayout: HorizontalLayout,
    HorizontalLayoutColumn: HorizontalLayoutColumn,
    VerticalLayout: VerticalLayout,
    VerticalLayoutRow: VerticalLayoutRow,
    HorizontalMenu: HorizontalMenu,
    HorizontalMenuImage: HorizontalMenuImage,
    HorizontalMenuLink: HorizontalMenuLink,
    HorizontalMenuColumn: HorizontalMenuColumn,
    Path: Path,
    PathItem: PathItem,
    LoadingMask: LoadingMask,
    Tree: Tree,
    TreeBranch: TreeBranch,
    TreeBranchlet: TreeBranchlet,
    TreeLeaf: TreeLeaf,
    TreeLeafColumn: TreeLeafColumn,
    TreeLinkButton: TreeLinkButton,
    TreePushButton: TreePushButton,
    Modal: Modal,
    Window: Window,
    Window_Props: Window_Props,
    WindowBody: WindowBody,
    WindowCloseButton: WindowCloseButton,
    WindowHead: WindowHead,
    ScrollArea: ScrollArea
};

export {

    // bundle of all components
    Zen,

    // button
    Button_Props,
    Combobox,
    ConjoinedButtons,
    LinkButton,
    PushButton,
    ToggleButton,
    ToggleRadioButton,
    
    // content
    Text,
    Text_Props,

    // input
    Checkbox,
    Checkbox_Props,
    NumberField,
    PasswordField,
    Radio,
    TextField,
    TimeField,

    // layout
    HorizontalLayout,
    HorizontalLayoutColumn,
    VerticalLayout,
    VerticalLayoutRow,

    // navigation
    HorizontalMenu,
    HorizontalMenuImage,
    HorizontalMenuLink,
    HorizontalMenuColumn,
    Path,
    PathItem,

    // panel
    LoadingMask,
    Tree,
    TreeBranch,
    TreeBranchlet,
    TreeLeaf,
    TreeLeafColumn,
    TreeLinkButton,
    TreePushButton,

    // window
    Modal,
    Window,
    Window_Props,
    WindowBody,
    WindowCloseButton,
    WindowHead,

    // container
    ScrollArea
}
