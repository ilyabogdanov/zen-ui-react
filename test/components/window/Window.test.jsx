jest.dontMock("jquery");
import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import $ from "jquery";
import Window from "../../../src/components/window/Window";
import WindowHead from "../../../src/components/window/WindowHead";
import WindowCloseButton from "../../../src/components/window/WindowCloseButton";
import WindowBody from "../../../src/components/window/WindowBody";
import Window_Props from "../../../src/components/window/Window_Props";
import { WINDOW_MESSAGES, WINDOW_HEAD_MESSAGES } from "../../../src/Messages";

class Container extends React.Component {
    constructor() {
        super();
        this._onClick = this._onClick.bind(this);
    }
    _onClick(e) {
        console.log(e);
    }
    render() {
        const { _onClick } = this;
        return (
            <Window>
                <WindowHead>
                    <WindowCloseButton onClick={_onClick}/>
                </WindowHead>
                <WindowBody/>
            </Window>
        )
    }
};

const div = document.createElement("div");
document.body.appendChild(div);

describe("Window", () => {

    // noinspection JSCheckFunctionSignatures
    const logSpy = jest.spyOn(console, "log");

    test("without children should throw error", () => {
        expect(() => {
            renderer.create(
                <Window/>
            );
        }).toThrowError(WINDOW_MESSAGES.NO_CHILDREN);
        expect(() => {
            renderer.create(
                <Window>
                    <WindowHead/>
                </Window>
            );
        }).toThrowError(WINDOW_MESSAGES.NO_CHILDREN);
        expect(() => {
            renderer.create(
                <Window>
                    <WindowBody/>
                </Window>
            );
        }).toThrowError(WINDOW_MESSAGES.NO_CHILDREN);
    });
    test("with illegal child should throw error", () => {
        expect(() => {
            renderer.create(
                <Window>
                    <div/>
                </Window>
            );
        }).toThrowError(WINDOW_MESSAGES.ILLEGAL_CHILD);
        expect(() => {
            renderer.create(
                <Window>
                    <WindowHead>
                        <div/>
                    </WindowHead>
                    <WindowBody/>
                </Window>
            );
        }).toThrowError(WINDOW_HEAD_MESSAGES.ILLEGAL_CHILD);
    });
    test("without palette should set ordinary palette", () => {

        const wrapper = mount(
            <Window>
                <WindowHead>
                    <WindowCloseButton onClick={function () {}}/>
                </WindowHead>
                <WindowBody/>
            </Window>, { attachTo: div }
        );
        expect(div.childNodes).toHaveLength(1);

        expect($(".zen_ui__window").attr("data-palette")).toEqual(Window_Props.palette.ORDINARY);
        expect($(".zen_ui__button").attr("data-palette")).toEqual(Window_Props.palette.ORDINARY);

        wrapper.detach();
        expect(div.childNodes).toHaveLength(0);

    });
    test("with palette, width and maxWidth should set them accordingly", () => {

        const wrapper = mount(
            <Window palette={Window_Props.palette.BRANDED} width="100%" maxWidth="620px">
                <WindowHead>
                    <WindowCloseButton onClick={function () {}}/>
                </WindowHead>
                <WindowBody/>
            </Window>, { attachTo: div }
        );
        expect(div.childNodes).toHaveLength(1);

        expect($(".zen_ui__window").attr("data-palette")).toEqual(Window_Props.palette.BRANDED);
        expect($(".zen_ui__window").css("width")).toEqual("100%");
        expect($(".zen_ui__window").css("max-width")).toEqual("620px");
        expect($(".zen_ui__button").attr("data-palette")).toEqual(Window_Props.palette.BRANDED);

        wrapper.detach();
        expect(div.childNodes).toHaveLength(0);

    });
    test("should execute callback when pushed on close button", () => {

        const component = mount(
            <Container/>,
            { attachTo: div }
        );
        expect(div.childNodes).toHaveLength(1);

        $(".zen_ui__button").click();
        expect(logSpy).toHaveBeenCalledTimes(1);

        component.detach();
        expect(div.childNodes).toHaveLength(0);
    });

    test("should render", () => {
        const e1 = renderer.create(
            <Window stretch={true}>
                <WindowHead>
                    <WindowCloseButton onClick={function () {}}/>
                </WindowHead>
                <WindowBody padding="1rem"/>
            </Window>
        ).toJSON();
        const e2 = renderer.create(
            <Window stretch={false}>
                <WindowHead>
                    <WindowCloseButton onClick={function () {}}/>
                </WindowHead>
                <WindowBody padding="1rem"/>
            </Window>
        ).toJSON();
        expect(e1).toMatchSnapshot();
        expect(e2).toMatchSnapshot();
    });
});
