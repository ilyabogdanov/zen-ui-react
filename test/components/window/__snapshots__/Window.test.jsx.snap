// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Window should render 1`] = `
<div
  className="zen_ui__window"
  data-palette="ordinary"
  data-stretch={true}
  style={
    Object {
      "maxWidth": null,
      "minWidth": null,
      "width": null,
    }
  }
>
  <div
    className="zen_ui__window_head"
  >
    <div
      className="zen_ui__stretch_margin"
    >
      <div
        className="zen_ui__horizontal_layout"
        style={
          Object {
            "height": "100%",
          }
        }
      >
        <div
          className="zen_ui__horizontal_layout_column"
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <div>
            <div>
              <span
                className="zen_ui__clipped_text"
              >
                <span
                  style={
                    Object {
                      "padding": "0 0.7rem",
                    }
                  }
                />
              </span>
            </div>
          </div>
        </div>
        <div
          className="zen_ui__horizontal_layout_column"
        >
          <div
            style={
              Object {
                "padding": "0.08rem 0.08rem 0.16rem 0.08rem",
                "width": "2.3rem",
              }
            }
          >
            <div>
              <div
                className="zen_ui__button_container"
                data-stretch="true"
              >
                <button
                  className="zen_ui__button"
                  data-palette="ordinary"
                  onClick={[Function]}
                >
                  <div
                    className="zen_ui__vector_icon__close"
                  />
                </button>
                <div />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="zen_ui__window_body"
    style={Object {}}
  >
    <div
      style={
        Object {
          "margin": "1rem",
        }
      }
    />
  </div>
</div>
`;

exports[`Window should render 2`] = `
<div
  className="zen_ui__window"
  data-palette="ordinary"
  data-stretch={false}
  style={
    Object {
      "maxWidth": null,
      "minWidth": null,
      "width": null,
    }
  }
>
  <div
    className="zen_ui__window_head"
  >
    <div
      className="zen_ui__stretch_margin"
    >
      <div
        className="zen_ui__horizontal_layout"
        style={
          Object {
            "height": "100%",
          }
        }
      >
        <div
          className="zen_ui__horizontal_layout_column"
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <div>
            <div>
              <span
                className="zen_ui__clipped_text"
              >
                <span
                  style={
                    Object {
                      "padding": "0 0.7rem",
                    }
                  }
                />
              </span>
            </div>
          </div>
        </div>
        <div
          className="zen_ui__horizontal_layout_column"
        >
          <div
            style={
              Object {
                "padding": "0.08rem 0.08rem 0.16rem 0.08rem",
                "width": "2.3rem",
              }
            }
          >
            <div>
              <div
                className="zen_ui__button_container"
                data-stretch="true"
              >
                <button
                  className="zen_ui__button"
                  data-palette="ordinary"
                  onClick={[Function]}
                >
                  <div
                    className="zen_ui__vector_icon__close"
                  />
                </button>
                <div />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="zen_ui__window_body"
    style={
      Object {
        "padding": "1rem",
      }
    }
  >
    <div
      style={Object {}}
    />
  </div>
</div>
`;
